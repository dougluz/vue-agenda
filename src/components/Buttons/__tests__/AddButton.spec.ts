import { mount } from "@vue/test-utils";
import { describe, expect, it } from "vitest";
import AddButtonVue from "../AddButton.vue";

describe("AddButton", () => {
  describe("when rendering the default version", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(AddButtonVue);
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should not contain the add-button-small class", () => {
      const wrapper = mount(AddButtonVue);
      expect(wrapper.classes()).not.toContain("add-button-small");
    });
  });

  describe("when rendering the small version", () => {
    it("should match the snapshot", () => {
      const wrapper = mount(AddButtonVue, {
        props: {
          small: true,
        },
      });
      expect(wrapper.html()).toMatchSnapshot();
    });

    it("should contain the add-button-small class", () => {
      const wrapper = mount(AddButtonVue, {
        props: {
          small: true,
        },
      });
      expect(wrapper.classes()).toContain("add-button-small");
    });
  });
});
