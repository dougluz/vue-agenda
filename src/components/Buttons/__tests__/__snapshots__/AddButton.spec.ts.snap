// Vitest Snapshot v1

exports[`AddButton > when rendering the default version > should match the snapshot 1`] = `
"<button class=\\"add-button-base\\" data-v-187b35fa=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" data-v-187b35fa=\\"\\">
    <path fill=\\"#FA7268\\" fill-rule=\\"nonzero\\" d=\\"M16 7.238H8.762V0H7.238v7.238H0v1.524h7.238V16h1.524V8.762H16z\\"></path>
  </svg> <span data-v-187b35fa=\\"\\">Criar contato</span></button>"
`;

exports[`AddButton > when rendering the small version > should match the snapshot 1`] = `
"<button class=\\"add-button-base add-button-small\\" data-v-187b35fa=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 16 16\\" data-v-187b35fa=\\"\\">
    <path fill=\\"#FA7268\\" fill-rule=\\"nonzero\\" d=\\"M16 7.238H8.762V0H7.238v7.238H0v1.524h7.238V16h1.524V8.762H16z\\"></path>
  </svg> <span data-v-187b35fa=\\"\\">Criar contato</span></button>"
`;
